src/box/lua/net_box.c

```diff
@@ -2554,7 +2564,8 @@ netbox_transport_process_requests(struct netbox_transport *transport,
 				  struct lua_State *L, uint32_t schema_version)
 {
 	if (transport->state != NETBOX_GRACEFUL_SHUTDOWN) {
-		assert(transport->state == NETBOX_FETCH_SCHEMA);
+		assert(transport->opts.fetch_schema == false ||
+			transport->state == NETBOX_FETCH_SCHEMA);
```

Comment:
I'd like to explicitly check the previous state here, no matter if `fetch_schema` is set or not. Let's change to

```C
assert(transport->state == NETBOX_AUTH ||
       transport->state == NETBOX_FETCH_SCHEMA);
```

Also, please mind indentation: we indent by parenthesis, using tabs first and then spaces.

Reply:
After resolving comment below it's possible that state is active. I changed assert to
```
assert(transport->state == NETBOX_ACTIVE ||
       transport->state == NETBOX_AUTH ||
       transport->state == NETBOX_FETCH_SCHEMA);
```

Reply:
However such approach triggers lua callback that is anyway is ignored because it's active->active state change

Reply:
Let's fix it like this instead:

```C
	if (transport->state != NETBOX_ACTIVE &&
	    transport->state != NETBOX_GRACEFUL_SHUTDOWN) {
		assert(transport->state == NETBOX_AUTH ||
		       transport->state == NETBOX_FETCH_SCHEMA);
```

Reply:
Yes, it's better. Fixed.